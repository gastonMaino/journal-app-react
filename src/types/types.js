export const types = {
    login: '[Auth] Login',
    logout: '[Auth] Logout',
    uiSetError: '[UI] Set Error',
    uiRemoveError: '[UI] Remove Error',
    uiStartLoading: '[UI] Start Loading',
    uiRemoveLoading: '[UI] Remove Loading',
    notesAddNew: '[NOTES] NEW NOTE',
    notesActive: '[NOTES] SET NOTE ACTIVE',
    notesLoad: '[NOTES] LOAD NOTE',
    notesUpdate: '[NOTES] UPDATE NOTE',
    notesFileUrl: '[NOTES] UPDATE IMAGE URL',
    notesDelete: '[NOTES] DELETE NOTE',
    notesLogoutCleaning: '[NOTES] LOGOUT CLEANING'

}