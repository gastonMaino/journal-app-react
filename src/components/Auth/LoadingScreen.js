import React from 'react'

export const LoadingScreen = () => {
    return (
        <div className='auth__register'>
            <div className='auth__register-preloader'></div>
            <p className='auth__register-text'>wait...</p>
        </div>
    )
}
