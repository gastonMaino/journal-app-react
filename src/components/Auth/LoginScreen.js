import React from 'react'
import { useDispatch, useSelector } from 'react-redux'
import { Link } from 'react-router-dom'
import { startGoogleLogin, startLoginEmailPassword } from '../../actions/auth'
import { useForm } from '../../hooks/useForm/useForm'

export const LoginScreen = () => {

    const dispatch = useDispatch();
    const { ui } = useSelector(state => state);
    const { loading } = ui;


    const [values, handleInputChange] = useForm({
        email: '',
        password: ''
    })

    const { email, password } = values;

    const handleLogin = (e) => {
        e.preventDefault();

        dispatch(startLoginEmailPassword(email, password))
    }

    const handleGoogleLogin = () => {
        dispatch(startGoogleLogin());
    }


    return (
        <>
            <h1 className='auth__title mb-4'>Login</h1>

            <form
                onSubmit={handleLogin}
                className='animate__animated animate__fadeIn animate__fast'
            >
                <input
                    type='text'
                    placeholder='Enter your email'
                    name='email'
                    value={email}
                    className='auth__input'
                    onChange={handleInputChange}
                />

                <input
                    type='password'
                    placeholder='Enter your password'
                    name='password'
                    value={password}
                    className='auth__input'
                    onChange={handleInputChange}
                />


                <button className='btn btn-primary btn-block' disabled={loading} type='submit'>
                    Login
                </button>

            </form>
            <div className='auth__social-networks animate__animated animate__fadeIn animate__fast'>
                <h2>Login with social networks</h2>
                <div className='google-btn' onClick={handleGoogleLogin}>
                    <div className="google-icon-wrapper">
                        <img className="google-icon" src="https://upload.wikimedia.org/wikipedia/commons/5/53/Google_%22G%22_Logo.svg" alt="google button" />
                    </div>
                    <p className="btn-text">
                        <b>Sign in with google</b>
                    </p>
                </div>
            </div>
            <Link className='link' to='/auth/register'>
                Create new account
            </Link>
        </>
    )
}
