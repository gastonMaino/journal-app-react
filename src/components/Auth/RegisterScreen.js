import React from 'react';
import validator from 'validator';
import { Link } from 'react-router-dom';
import { useDispatch, useSelector } from 'react-redux';

import { useForm } from '../../hooks/useForm/useForm';
import { removeError, setError } from '../../actions/ui';
import { startRegisterWithEmail } from '../../actions/auth';
// import { startLoginEmailPassword } from '../../actions/auth';

export const RegisterScreen = () => {

    const dispatch = useDispatch();
    const { msgError } = useSelector(state => state.ui);


    const [values, handleInputChange] = useForm({
        name: '',
        email: '',
        password: '',
        password2: ''
    })

    const { name, email, password, password2 } = values;

    const handleRegister = (e) => {
        e.preventDefault();

        if (isFormValid()) {
            dispatch(startRegisterWithEmail(name, password, email));
        }
    }

    const isFormValid = () => {

        if (name.length === 0) {
            dispatch(setError('name is required'));
            return false;
        } else if (!validator.isEmail(email)) {
            dispatch(setError('email is no valid'));
            return false;
        } else if (password !== password2 || password.length < 5) {
            dispatch(setError('password should be at 6 characters and match each other'));
            return false
        }

        dispatch(removeError());
        return true;
    }

    return (
        <>
            <h1 className='auth__title mb-4'>Register</h1>

            <form
                onSubmit={handleRegister}
                className='animate__animated animate__fadeIn animate__fast'
            >
                {
                    msgError
                    &&
                    <div className='auth__alert-error'>{msgError}</div>
                }
                <input
                    type='text'
                    placeholder='Name'
                    name='name'
                    value={name}
                    className='auth__input mb-3'
                    onChange={handleInputChange}
                />

                <input
                    type='text'
                    placeholder='Email'
                    name='email'
                    value={email}
                    className='auth__input mb-3'
                    onChange={handleInputChange}
                />

                <input
                    type='password'
                    placeholder='Password'
                    name='password'
                    value={password}
                    className='auth__input mb-3'
                    onChange={handleInputChange}
                />
                <input
                    type='password'
                    placeholder='Confirm password'
                    name='password2'
                    value={password2}
                    className='auth__input mb-3'
                    onChange={handleInputChange}
                />

                <button className='btn btn-primary btn-block' type='submit'>
                    Register
                </button>

            </form>

            <Link className='link mt-2' to='/auth/login'>
                Already register ?
            </Link>
        </>
    )
}
