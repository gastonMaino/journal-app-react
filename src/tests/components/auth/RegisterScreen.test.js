import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import { mount } from 'enzyme/build';
import '@testing-library/jest-dom';

import { RegisterScreen } from '../../../components/Auth/RegisterScreen';
import { types } from '../../../types/types';

const middlewares = [thunk];
const mockStore = configureStore(middlewares);
const initState = {
    auth: {},
    ui: {
        loading: false,
        msgError: null
    },
    notes:{
        notes:[],
        active: null
    }
};

const store = mockStore(initState);

const wrapper = mount(
    <Provider store={store}>
        <MemoryRouter>
            <RegisterScreen />
        </MemoryRouter>
    </Provider>
);


describe('test in RegisterScreen', () => {

    test('should match to snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    })

    test('should make dispatch', () => {

        wrapper.find('form').simulate('submit', {
            preventDefault(){}
        })

        const actions = store.getActions();
        expect(actions[0]).toEqual({
            type: types.uiSetError,
            payload: 'name is required'
        })  
    })

    test('should  show alert box', () => {
        const initState = {
            auth: {},
            ui: {
                loading: false,
                msgError: 'name is required'
            },
            notes:{
                notes:[],
                active: null
            }
        };
        
        const store = mockStore(initState);
        
        const wrapper = mount(
            <Provider store={store}>
                <MemoryRouter>
                    <RegisterScreen />
                </MemoryRouter>
            </Provider>
        );
        
        expect(wrapper.find('.auth__alert-error').exists()).toBe(true)
        expect(wrapper.find('.auth__alert-error').text().trim()).toBe(initState.ui.msgError);

    })
    
    
    
})
