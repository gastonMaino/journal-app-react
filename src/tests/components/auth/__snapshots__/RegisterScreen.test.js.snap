// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test in RegisterScreen should match to snapshot 1`] = `
Array [
  <h1
    className="auth__title mb-4"
  >
    Register
  </h1>,
  <form
    className="animate__animated animate__fadeIn animate__fast"
    onSubmit={[Function]}
  >
    <input
      className="auth__input mb-3"
      name="name"
      onChange={[Function]}
      placeholder="Name"
      type="text"
      value=""
    />
    <input
      className="auth__input mb-3"
      name="email"
      onChange={[Function]}
      placeholder="Email"
      type="text"
      value=""
    />
    <input
      className="auth__input mb-3"
      name="password"
      onChange={[Function]}
      placeholder="Password"
      type="password"
      value=""
    />
    <input
      className="auth__input mb-3"
      name="password2"
      onChange={[Function]}
      placeholder="Confirm password"
      type="password"
      value=""
    />
    <button
      className="btn btn-primary btn-block"
      type="submit"
    >
      Register
    </button>
  </form>,
  <a
    className="link mt-2"
    href="/auth/login"
    onClick={[Function]}
  >
    Already register ?
  </a>,
]
`;
